nestjs/src/courses/course-mentors/dto/mentor-details.dto.ts

```diff
@@ -0,0 +1,58 @@
+import { PreferredStudentsLocation } from '@common/enums/mentor';
+import { InterviewStatistics, MentorDetails } from '@common/models';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { StudentDto } from '../../../courses/students/dto';
+import { UserDto } from '../../../user-groups/dto';
+
+export class MentorDetailsDto extends UserDto implements MentorDetails {
+  constructor(mentor: MentorDetails) {
+    super(mentor);
+
+    this.isActive = mentor.isActive;
+    this.cityName = mentor.cityName;
+    this.countryName = mentor.countryName;
+    this.students = mentor.students;
+    this.interviews = mentor.interviews;
+    this.screenings = mentor.screenings;
+    this.taskResultsStats = mentor.taskResultsStats;
+    this.maxStudentsLimit = mentor.maxStudentsLimit;
+    this.studentsPreference = mentor.studentsPreference as PreferredStudentsLocation;
+    this.studentsCount = mentor.studentsCount;
+  }
+
+  @ApiProperty()
+  isActive: boolean;
+
+  @ApiProperty()
+  cityName: string;
+
+  @ApiProperty()
+  countryName: string;
+
+  @ApiProperty()
+  maxStudentsLimit: number;
+
+  @ApiProperty({ type: [StudentDto] })
+  students: (StudentDto | { id: number })[];
```

Comment:
looks suspicious: why we can get both? IMO would be better to have one possible option here, especially since you specifically Property as Student

Reply:
It's because I specify that DTO should implement the MentorDetails interface and the MentorBasic interface [has defined that field in such a way](https://github.com/rolling-scopes/rsschool-app/blob/109fc2431ce4074ec7fd310504d019cdfdccc3be/common/models/user.ts#L13).
Should I delete `implement MentorDetails` and specify one type?

Reply:
I assume there is not need to implement it. before it was a shared type, but once you move everything to nestjs, our types are generated. So no need to add deps on `common`

Reply:
> I assume there is not need to implement it. before it was a shared type, but once you move everything to nestjs, our types are generated. So no need to add deps on `common`

Got it, thank you

Reply:
Fixed